Wrappers/Python/cil/utilities/display.py

```diff
@@ -92,6 +93,205 @@ def save(self,filename, **kwargs):
         except:
             print("Unable to save image")
 
+
+class show1D(show_base):
+    """
+    This creates 1D plots of pixel values by slicing multi-dimensional
+    data.
+
+    Parameters
+    ----------
+    data : DataContainer, list of DataContainer, tuple of DataContainer
+        Multi-dimensional data to be reduced to 1D.
+    line_coords : list of tuples, default=None
+        (dimension, coordinate) pairs for slicing `data`
+    label : str, list of str, default=None
+        Label(s) to use in the plot's legend
+    title : str, default None
+        A title for the plot
+    color : str, list of str, default=None
+        Color(s) for each line plot
+    size : tuple, default=(8,6)
+        The size of the figure
+    axis_labels : tuple of str, list of str, default=('Pixel','Pixel
+    value')
+        Axis labels in the form (x_axis_label,y_axis_label)
+    force : bool, default=True
+        Passed to `get_slice`
+
+    Attributes
+    ----------
+    figure : matplotlib.figure.Figure
+
```

Comment:
So that it's easier to understand how to use it, maybe you could add an Examples section with an example of a show1D call so you can demonstrate the use of the tuples etc.

Reply:
I hadn't quite understood how to construct the list of tuples and whether I was able to make multiple lines on the same plot so maybe a little more explanation somewhere might help.

As we discussed, I saw that you can have 2 lines for different datasets at the same coordinates e.g. compare 1D plot for angle 80 and vertical 40 on 2 datasets.

However at the moment you are unable to plot 2 lines for the same dataset at different coordinates. However, I don't know if that would be a useful feature anyway 🤔

Reply:
I've added a couple of examples to the `show1d` docstring - do you think they're sufficient?

Reply:
Examples are great, thanks!

My understanding now is that if you provide 2 datasets you get 2 separate graphs whereas if you provide a single dataset but multiple slices you get them on the same graph? 

I wonder if it would be better to allow the user to choose if they would like it on the same or different graphs in these cases, or am I over-complicating things?

Reply:
The behaviour should be somewhat opposite to that:

- if you provide a single dataset, and a single set of slices, then you should get a single graph with a line plot
- if you provide multiple datasets and a single set of slices, you get a single graph with multiple line plots on it (one for each dataset)
- if you provide a single dataset and multiple sets of slices, you get multiple graphs (one for each line defined by the slices), each with a single line plot
- and with multiple datasets and multiple sets of slices, you get one graph for each 1D slice (defined by the slices) with multiple line plots on it.

The idea here was that for a given 1D slice, you probably(?) want to compare that slice across datasets as opposed to, say, comparing several line plots for the same data on the same graph.

There are many different ways that this data could be plotted, I think - if it would be useful to plot all the data on a single graph, we could add that as an option.

Reply:
It is the opposite, sorry I wrote the wrong thing 😆

Reply:
Adding this info to the documentation would be a good idea I think

Reply:
I've added a more succinct version of this to the docs in the most recent commit 😄

Reply:
thanks! Now I will know what it does when I come back to it in a month or so, having forgotten 😄